ironfish/src/rpc/routes/router.ts

```diff
@@ -52,42 +52,18 @@ export function parseRoute(
 }
 
 export class Router {
-  routes = new Map<string, Map<string, { handler: RouteHandler; schema: YupSchema }>>()
-  context: RequestContext = {}
-  server: RpcServer | null = null
-
-  register<TRequestSchema extends YupSchema, TResponse>(
-    route: string,
-    requestSchema: TRequestSchema,
-    handler: RouteHandler<YupSchemaResult<TRequestSchema>, TResponse>,
-  ): void {
-    const [namespace, method] = parseRoute(route)
-
-    Assert.isNotUndefined(namespace, `Invalid namespace: ${String(namespace)}: ${route}`)
-    Assert.isNotUndefined(method, `Invalid method: ${String(namespace)}: ${route}`)
-
-    let namespaceRoutes = this.routes.get(namespace)
-
-    if (!namespaceRoutes) {
-      namespaceRoutes = new Map<string, { handler: RouteHandler; schema: YupSchema }>()
-      this.routes.set(namespace, namespaceRoutes)
-    }
+  routes = new Routes()
+  server: RpcServer
 
-    namespaceRoutes.set(method, {
-      handler: handler as RouteHandler<unknown, unknown>,
-      schema: requestSchema,
-    })
+  constructor(routes: Routes, server: RpcServer) {
+    this.routes = routes
```

Comment:
should this be a copy? Or else any modifications to this router modifies the root router permanently. Like in the tests in these files.

Reply:
`Router` is only instantiated in one place right now (Server) and it uses filter before passing in the routes. It could be good for safety but prob not needed yet unless we start doing more complicated things with the Router. Just gonna assume whoever is instantiating it thought of that